src/metabase/native_query_analyzer.clj

```diff
@@ -0,0 +1,55 @@
+(ns metabase.native-query-analyzer
+  "Integration with Macaw, which parses native SQL queries. All SQL-specific logic is in Macaw, the purpose of this
+  namespace is to:
+
+  1. Translate Metabase-isms into generic SQL that Macaw can understand.
+  2. Contain Metabase-specific business logic."
+  (:require
+   [clojure.string :as str]
+   [macaw.core :as mac]
+   [metabase.util :as u]
+   [toucan2.core :as t2]))
+
+(defn- normalize-name
+  ;; TODO: This is wildly naive and will be revisited once the rest of the plumbing is sorted out
+  [name]
+  (-> name
+      (str/replace "\"" "")
+      u/lower-case-en))
+
+(defn- fields-for-query
+  [q db-id]
+  (let [{:keys [columns tables]} (-> q
+                                     mac/parsed-query
+                                     mac/query->components)]
+    (t2/query {:select [[:f/id :field_id] [:f/name :column_name] [:t/name :table_name]]
```

Comment:
I was expecting us to do some kind of filtering here where we resolve ambiguous columns. Is that idea that we'll just track the union of all things that the query could possibly refer to?

It's not clear to me from this code whether we handle the non-ambiguous cases where columns are qualified - unit tests covering all these edges would be great!

Since there are no bad ideas in a brainstorm, in the extreme case we could shrink the query or use a cache of its results to reason about which fields it used. Sounds like this would break horrifically for complex queries, but you know what they say about 80% 😜

Reply:
> Is that idea that we'll just track the union of all things that the query could possibly refer to?

For now. I wanted to get the basic plumbing of this PR out the door. The next step is making things smarter along the lines you described, but I wanted to separate the concerns of making new app DB tables and API endpoints and such from twiddling around with table inference.